### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by seadowg**
Some updates to our docs based (and one to the default gradle config). I've left comments inline on hopefully justifying any changes I've made but these are all very open for discussion.

### CODE REVIEW

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:58]: This is part of the PR template tasks and doesn't feel to me like it needs to be highlighted here

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:66]: Again this is in the PR template tasks

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:73]: We don't require two people as far as I've seen. Maybe we should add that it requires a review from a committer? I guess this is kind of implicit given a committer is required to merge.

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:83]: We don't use the `needs review` label anymore. Guessing Github has got better at flagging this? I definitely don't feel like it's something we need to be doing - any PR that doesn't have a review needs reviewed.

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:85]: Got rid of this seeing as I removed the need for a second review.

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:86]: Added this to make it explicit that CI has to be green for us to merge. Also thought it was good to specify that only committers can merge and get a mention of Code Freeze in here (currently only in `README.md`).

File: docs/CONTRIBUTING.md
Comment (seadowg) [inline docs/CONTRIBUTING.md:91]: Github fixed this 🎉

File: docs/STATE.md
Comment (seadowg) [inline docs/STATE.md:19]: Android docs now have clearer language on this split so wanted to reflect that here.

File: docs/STATE.md
Comment (seadowg) [inline docs/STATE.md:47]: We haven't talked about this, but it's definitely the direction we're going in

File: gradle.properties
Comment (seadowg) [inline gradle.properties:5]: Realized people might not be benefiting from this. This is paying more and more dividends as we commit more to multimodule as each one can have tasks run on them in parallel.

**Review by lognaturel — APPROVED** [review]
Thank you!!

**Review by grzesiek2010 — APPROVED** [review]

**Pull request merged**
